share/util/__init__.py

```diff
@@ -124,7 +124,7 @@ def __visit(self, node):
 
         self.__visiting.add(key)
         for k in self.__dependencies(node):
-            if k is not None:
+            if k is not None and k is not key:
```

Comment:
That's to support bepress subjects with `central_synonym` pointing at itself. It's replaced with `None` in `Subject.normalize`, but ChangeGraph sorts itself on init, before calling normalize, and will break here.

...that said, if the OSF side [is indeed switching to `None`](https://github.com/CenterForOpenScience/osf.io/pull/7243#discussion_r119704274), this is unnecessary.